src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -98,33 +105,90 @@ public static Checksum Create(IObjectWritable @object)
 
         public static Checksum Create(Checksum checksum1, Checksum checksum2)
         {
-            using var stream = SerializableBytes.CreateWritableStream();
+#if NET
+            return CreateUsingSpans(checksum1, checksum2);
+#else
+            return CrateUsingByteArrays(checksum1, checksum2);
+#endif
+        }
 
-            using (var writer = new ObjectWriter(stream, leaveOpen: true))
-            {
-                checksum1.WriteTo(writer);
-                checksum2.WriteTo(writer);
-            }
+        public static Checksum Create(Checksum checksum1, Checksum checksum2, Checksum checksum3)
+        {
+#if NET
+            return CreateUsingSpans(checksum1, checksum2, checksum3);
+#else
+            return CrateUsingByteArrays(checksum1, checksum2, checksum3);
+#endif
+        }
 
-            stream.Position = 0;
-            return Create(stream);
+        private static Checksum CrateUsingByteArrays(Checksum checksum1, Checksum checksum2)
```

Comment:
Should the byteArray methods be in a #else for a #if NET?

Reply:
if i do that, i can't then write a test that validates them against each other... (Definitely looking for clever ideas here.  Just not sure what they are!)